geograypher/cameras/derived_cameras.py

```diff
@@ -148,6 +148,53 @@ def __init__(
             local_to_epsg_4978_transform=chunk_to_epsg4978,
         )
 
+    def ideal_to_warped(
+        self, camera: PhotogrammetryCamera, xpix: np.ndarray, ypix: np.ndarray
+    ) -> typing.Tuple[np.ndarray, np.ndarray]:
+        """
+        For consistency, this should fully match the docstring from
+        cameras.PhotogrammetryCameraSet.ideal_to_warped()
+        """
+
+        # Convert from x and y pixels to the homogeneous camera frame
+        # Note that for the math to work out, the cx and cy terms are neglected in this step and
+        # only applied at the very end
+        principal_x = camera.image_width / 2.0
+        principal_y = camera.image_height / 2.0
+        x = (xpix - principal_x) / camera.f
+        y = (ypix - principal_y) / camera.f
+
+        # Enforce that a strict subset of expected parameters are found
+        params = sorted(camera.distortion_params.keys())
+        if not set(params) <= set(["b1", "b2", "k1", "k2", "k3", "k4", "p1", "p2"]):
+            raise ValueError(f"Unexpected distortion params found: {params}")
+        b1 = camera.distortion_params.get("b1", 0)
+        b2 = camera.distortion_params.get("b2", 0)
+        k1 = camera.distortion_params["k1"]  # Enforce that the most basic is required
```

Comment:
I'm not sure it's guaranteed we'd get this parameter. You could have a check to return the identity mapping if all parameters are 0 though.

Reply:
Hm. That's an interesting point, should this fail at that point? Basically the user has requested dewarping and there is not saved data to enable that. I think this should fail instead of just silently passing back an identity mapping and having the user think it's been dewarped.

Reply:
I see your logic and am fine with either option. I was thinking you might run this on 100 datasets in a batch and for a handful Metashape would not have warp parameters because they had a limited number of images. So it would be nice to be able to use the same config for all of them.

Reply:
I think it'd be dangerous to run on 100 datasets and assume their outputs are aligned with the warped images when they aren't always, you'd have potentially a few missions with mismatched data polluting things. We can work out some sort of system where if we can't warp the pix2face output then we don't warp it and mark it as such. If that's coming up soon we can make a followup PR right after this

Reply:
Fair. I think I assumed that if given parameters weren't estimated that the modeling worked well without them, so neglecting distortion modeling would be fine in those cases. But I have nothing to back that up. Erroring out seems like a good solution for now. I don't think this will be a problem for us internally in the near future because automate-metashape is configured to estimate `k1` and `k2` for all datasets as far as I understand it.